go-controller/pkg/ovn/policy.go

```diff
@@ -700,6 +706,24 @@ func (oc *Controller) handlePeerNamespaceSelectorModify(
 	}
 }
 
+// we do not need to create an address set
+// - if there is 'zero' NetworkPolicyPeer entity
+// - if the number of IPBlock elements is same as the number of NetworkPolicyPeer entities
```

Comment:
while this is technically true, I think the version of the logic in the earlier `hasAnyLabelSelector` made more sense; you must create an address set if any peer has a `PodSelector` or a `NamespaceSelector`

Reply:
@danwinship I agree. I have moved to `hasAnyLabelSelector`, so PTAL.